insights/core/__init__.py

```diff
@@ -780,21 +780,28 @@ class JSONParser(Parser, LegacyItemAccess):
     A parser class that reads JSON files.  Base your own parser on this.
     """
     def parse_content(self, content):
+        # If content is empty then raise a skip exception instead of a parse exception.
+        if not content:
+            raise SkipComponent("Empty output.")
         try:
             if isinstance(content, list):
-                self.data = json.loads('\n'.join(content))
+                # Find the actual json start line with '{' and '[' as identifier
+                # To skip any extra lines before the actual json start line
+                actual_start_index = 0
+                for idx, _line in enumerate(content):
+                    line = _line.strip()
+                    if line and line.startswith('{') or line.startswith('['):
+                        actual_start_index = idx
+                        break
+                self.data = json.loads('\n'.join(content[actual_start_index:]))
```

Comment:
It would be good to keep these skipped lines to an e.g. `unparsed_lines` for further usage.

Reply:
@xiangce Updated, please kindly help review.